kairos-contracts/demo-contract/contract/src/main.rs

```diff
@@ -8,23 +8,24 @@ use casper_contract::{
     unwrap_or_revert::UnwrapOrRevert,
 };
 use casper_event_standard::Schemas;
+use casper_types::bytesrepr::Bytes;
 use casper_types::{
     contracts::NamedKeys, runtime_args, AccessRights, ApiError, CLValue, EntryPoints, Key,
     RuntimeArgs, URef, U512,
 };
 use contract_utils::constants::{
     KAIROS_CONTRACT_HASH, KAIROS_CONTRACT_PACKAGE_HASH, KAIROS_CONTRACT_UREF, KAIROS_DEPOSIT_PURSE,
-    KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER, RUNTIME_ARG_AMOUNT, RUNTIME_ARG_TEMP_PURSE,
+    KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER, KAIROS_TRIE_ROOT, RUNTIME_ARG_AMOUNT,
+    RUNTIME_ARG_INITIAL_TRIE_ROOT, RUNTIME_ARG_RECEIPT, RUNTIME_ARG_TEMP_PURSE,
 };
 use contract_utils::Deposit;
 mod entry_points;
 mod utils;
+use risc0_zkvm::Receipt;
 use utils::errors::DepositError;
 use utils::get_immediate_caller;
 
-#[allow(clippy::single_component_path_imports)]
-#[allow(unused)]
-use casper_contract_no_std_helpers;
```

Comment:
This should not be removed in contract, not in session codes as well. If you are getting error about `panic_impl` duplicate, then it means you introduced `std` dependency, which unnecessarily increase WASM size.

Reply:
`casper_contract_no_std_helpers` is incompatible with risc0.

It looks like risc0 is not actually `no_std`.
`cargo build --no-default-features --target thumbv7m-none-eabi -p contract`
```
error[E0463]: can't find crate for `core`
  |
  = note: the `thumbv7m-none-eabi` target may not be installed
  = help: consider downloading the target with `rustup target add thumbv7m-none-eabi`

For more information about this error, try `rustc --explain E0463`.
error: could not compile `subtle` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `rand_core` (lib) due to 1 previous error
error: could not compile `cfg-if` (lib) due to 1 previous error
error: could not compile `const-oid` (lib) due to 1 previous error
error: could not compile `rand_core` (lib) due to 1 previous error
error: could not compile `hex` (lib) due to 1 previous error
error: could not compile `pin-project-lite` (lib) due to 1 previous error
error: could not compile `either` (lib) due to 1 previous error
error: could not compile `zeroize` (lib) due to 1 previous error
error: could not compile `ppv-lite86` (lib) due to 1 previous error
error: could not compile `static_assertions` (lib) due to 1 previous error
error: could not compile `tracing-core` (lib) due to 1 previous error
error: could not compile `num-traits` (lib) due to 1 previous error
error: could not compile `typenum` (lib) due to 1 previous error
error: could not compile `zerocopy` (lib) due to 1 previous error
error: could not compile `serde` (lib) due to 1 previous errorc
```

Reply:
This is not fixable right now.

Reply:
Maybe with the groth16 verifier, we can get no_std.

Reply:
We can try build-std in the future as well https://github.com/johnthagen/min-sized-rust

Reply:
```
error[E0463]: can't find crate for `core`
  |
  = note: the `thumbv7m-none-eabi` target may not be installed
  = help: consider downloading the target with `rustup target add thumbv7m-none-eabi`
```

That's not a proof that Risc0 is not `no_std`, just target is missing - you can install it with `rustup target add thumbv7m-none-eabi`.

Reply:
I did that first.